pyte/streams.py

```diff
@@ -18,15 +18,22 @@
                     see AUTHORS for details.
     :license: LGPL, see LICENSE for more details.
 """
+from __future__ import annotations
 
 import codecs
 import itertools
 import re
 import warnings
 from collections import defaultdict
+from typing import Any, Callable, Generator, Union, TYPE_CHECKING
 
 from . import control as ctrl, escape as esc
 
+if TYPE_CHECKING:
+    from .screens import Screen
+
+
+ParserGenerator = Generator[Union[bool, None], str, None]
```

Comment:
I personally find Optional[None] more readable tbh. Is there a reason why you prefer Union[T, None]?

Reply:
Personally I like `Union[x, None]` better since it's analogous to `x | None`.  I also think `Optional[x]` is only meaningful as a keyword arg like `def foo(x: Optional[int] = None)` since it's a strange word to use for a variable.  Like, in this instance, is the yield value of the generator "Optional"?  Well, no, it still needs to yield `None`.

But, I can change it to your preference no problem.

Reply:
I changed all Union[x, None] to Optional[x]